sql/core/src/test/scala/org/apache/spark/sql/DatasetSuite.scala

```diff
@@ -783,6 +783,16 @@ class DatasetSuite extends QueryTest with SharedSQLContext {
       ds.filter(_.b > 1).collect().toSeq
     }
   }
+
+  test("transformed dataset correctly solve the attributes") {
```

Comment:
can you double check this test? I tried this on latest master branch and it passed...